src/main/scala/beam/agentsim/agents/vehicles/HumanBodyVehicle.scala

```diff
@@ -1,50 +1,79 @@
+    cap.setSeats(1)
+    cap.setStandingRoom(0)
+    t.setCapacity(cap)
+    t
+  }
 }
 
 /**
-  *
   * @param weight in kilo
   * @param height in kilo
   */
 case class HumanDimension(weight: Double, height: Double) extends Dimension
 
-case class HumanBodyVehicleData(personId: Id[PersonAgent], dim: HumanDimension) extends VehicleData {
-  private lazy val humanBodyVehicleType = initVehicleType()
+case class HumanBodyVehicleData() extends BeamAgentData
 
-  private def initVehicleType() = {
-    val t  = VehicleUtils.getFactory.createVehicleType(Id.create("HumanBodyVehicle", classOf[VehicleType]))
-    val cap = VehicleUtils.getFactory.createVehicleCapacity()
-    cap.setSeats(1)
-    cap.setStandingRoom(1)
-    t.setCapacity(cap)
-    t
+object HumanBodyVehicle extends BeamVehicleObject{
+  //TODO make HumanDimension come from somewhere
+
+  // This props has it all
+  def props(beamServices: BeamServices, vehicleId: Id[Vehicle], personId: Id[PersonAgent], data: HumanBodyVehicleData, powerTrain: Powertrain,
+            initialMatsimVehicle: Vehicle, initialMatsimAttributes: Attributes) = {
+    Props(classOf[HumanBodyVehicle], beamServices, vehicleId, personId, data, powerTrain, initialMatsimVehicle, initialMatsimAttributes)
   }
 
-  override def vehicleTypeName: String = "HumanBodyVehicle"
+  // This props follows spec of BeamVehicle
+  override def props(beamServices: BeamServices, vehicleId: Id[Vehicle], matSimVehicle: Vehicle, powertrain: Powertrain): Props = {
+    val personId = Id.create("EMPTY",classOf[PersonAgent])
+    props(beamServices, vehicleId, personId, HumanBodyVehicleData(), powertrain, matSimVehicle, new Attributes())
+  }
 
-  override def vehicleClassName: String = null
+  // This props is specifically for vehicle creation during initialization
+  def props(beamServices: BeamServices, matSimVehicle: Vehicle, personId: Id[PersonAgent], powertrain: Powertrain): Props = {
+    props(beamServices, matSimVehicle.getId, personId, HumanBodyVehicleData(), powertrain, matSimVehicle,  new Attributes())
+  }
 
-  override def getType: VehicleType = humanBodyVehicleType
+  def PowertrainForHumanBody(): Powertrain = Powertrain.PowertrainFromMilesPerGallon(360) // https://en.wikipedia.org/wiki/Energy_efficiency_in_transport#Walking
 
-  override def getId: Id[Vehicle] = Id.create(personId.toString, classOf[Vehicle])
-}
+  def createId(personId: Id[Person]) : Id[Vehicle] = {
+    Id.create("body-" + personId.toString, classOf[Vehicle])
+  }
 
-object HumanBodyVehicle {
-  //TODO make HumanDimension come from somewhere
-  def props(services: BeamServices, personId: Id[PersonAgent]) = Props(classOf[HumanBodyVehicle],services,
-    HumanBodyVehicleData(personId, HumanDimension(1.7, 60.0)), new Trajectory(), None)
+  def isHumanBodyVehicle(beamVehicleId: Id[Vehicle]) = beamVehicleId.toString.toLowerCase.contains("body")
+  val placeHolderBodyVehilceId: Id[Vehicle] = Id.create("body",classOf[Vehicle])
```

Comment:
You're helping me change my mind on this. 

My original thinking was that since we're already treating HumanBody and walking as a special case during traveling, we can avoid the extra overhead of passing the BodyVehicleId from the person through the RoutingRequest only to be folded into the EmbodiedBeamTrip and passed back to the person.... instead we just use a dummy placeholder that the Person can look for and substitute their body Id.... but now that I've completed the extra work of folding car Id and bike Id into the RoutingResponse, I'm realizing that passing the Body vehicle doesn't really add anything and would remove the confusion of this dummy Id.... so I will change as a TODO.